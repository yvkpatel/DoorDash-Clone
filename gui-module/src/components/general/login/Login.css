.login-form
{
    text-align: center;
    /* background-color: whitesmoke; */
    border-radius: 10px;
    padding:10px;
    margin: 50px;
    align-items: 'center';
}

.login-text
{
    text-align: center;
    color: red;
    font-size: 40px;
}

.btn-primary
{
    font-size: 20px;
    padding: 10px 40px;
    margin: 10px;
    background-color: #2DB6F0 !important;
    color: white !important;
    border-color:  #2DB6F0 !important;
    align-items: 'center';
}

.or
{
    font-size:20px;
    font-family: Georgia, 'Times New Roman', Times, serif;
    text-align: center;
}

.form-group
{
    font-family:Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
    width:40%;
    max-width: 250px;
    border-radius: 5px;
    padding: 10px 10px;
    align-items: 'center';
}

.register-buttons
{
    text-align: center;
    color: white;
    /* background-color: whitesmoke; */
    border-radius: 10px;
    padding:10px;
    margin: 50px;
}

a
{
    text-decoration: none;
}