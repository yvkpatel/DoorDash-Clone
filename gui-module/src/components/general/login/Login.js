/**
 * This is the login screen
 *
 * Users of all types can login here. They will be redirected based on their user details
 */
import { React, useState } from "react";
import "./Login.css";
import Button from "react-bootstrap/Button";
import GenHeader from "../../GenHeader.js";
import LoginForm from "./LoginForm";
import { useHistory } from "react-router-dom";

const Login = () => {
  const [authenticate, setAuthenticate] = useState("false"); //true when user exists in database
  const [userType, setUserType] = useState(""); //customer, driver, restuarant
  const [userID, setUserID] = useState("");
  const history = useHistory();

  //Handles the submission of login details
  const submitLogin = (details) => {
    var obj = new Object();
    obj.username = details.email;
    obj.passHash = details.password;
    //testing purposes
    console.log(details);

    fetchAuthentication(obj);
    setUserType(details.userType);

    //Determine what to do based on the user type
    if (userType == "Customer") {
      fetchCustomer(details.email);
      history.push("/customer-dashboard");
      return;
    } else if (userType == "Restaurant") {
      fetchRestauant(details.email);
      history.push("/restaurant-dash");
      return;
    } else if (userType == "Driver") {
      fetchDriver(details.email);
      history.push("/driver-dash");
      return;
    }
  };

  //Calls database to get the authenticate the user
  //TODO: CURRENTLY NOT WORKING TO AUTHENTICATE USER - ASSUME TRUE FOR ALL CALLS FOR MVP
  const fetchAuthentication = (obj) => {
    setAuthenticate("true");
    // const reqPar = {
    //   method: 'POST',
    //   headers: { 'Content-Type': 'application/json' },
    //   body: JSON.stringify(obj)
    // }
    // console.log(reqPar.body);
    // fetch("/", reqPar).then(res => res.json())
    //   .then(data => setAuthenticate(data.reply))
    //   .catch(err => console.log(err));
  };

  //Gets Customer Details
  const fetchCustomer = (username) => {
    const res = fetch("/customer/user/" + username)
      .then((resp) => resp.json())
      .then((data) => setUserID(data._Customer__id))
      .catch((err) => console.log(err));
  };
  //Gets Restaurant Details
  const fetchRestauant = (username) => {
    const res = fetch("/restaurant/user/" + username)
      .then((resp) => resp.json())
      .then((data) => setUserID(data._Restaurant__id))
      .catch((err) => console.log(err));
  };
  //Gets Driver Details
  const fetchDriver = (username) => {
    const res = fetch("/driver/user/" + username)
      .then((resp) => resp.json())
      .then((data) => setUserID(data._Driver__id))
      .then(history.push("/customer-dashboard", { state: userID }))

      .catch((err) => console.log(err));
  };
  return (
    <div className="app" data-testid="login-1">
      <GenHeader></GenHeader>
      <div
        style={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
        }}
      >
        <LoginForm onLogin={submitLogin} />
      </div>

      <div className="or">
        <p>
          <strong>OR</strong>
        </p>
        <p className="signupText">Sign up as</p>
      </div>
      <div className="register-buttons">
        <Button
          href="/customer-signup"
          variant="primary"
          data-testid="customer-signup"
        >
          Customer
        </Button>
        &nbsp;
        <Button
          href="/driver-signup"
          variant="primary"
          data-testid="driver-signup"
        >
          Driver
        </Button>
        &nbsp;
        <Button
          href="/restaurant-regis"
          variant="primary"
          data-testid="restaurant-signup"
        >
          Restaurant
        </Button>
      </div>
    </div>
  );
};

export default Login;
