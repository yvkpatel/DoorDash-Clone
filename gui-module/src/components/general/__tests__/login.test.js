import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Login from "../login/Login";
import "@testing-library/jest-dom/extend-expect";

test("should render login component", () => {
  render(<Login />);
  const loginElement = screen.getByTestId("login-1");
  expect(loginElement).toBeInTheDocument();
});

test("should show three buttons", () => {
  render(<Login />);
  const customerElement = screen.getByTestId("customer-signup");
  const driverElement = screen.getByTestId("driver-signup");
  const restaurantElement = screen.getByTestId("restaurant-signup");
  expect(customerElement).toBeInTheDocument();
  expect(driverElement).toBeInTheDocument();
  expect(restaurantElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer.create(<Login />);
  expect(tree).toMatchSnapshot();
});
