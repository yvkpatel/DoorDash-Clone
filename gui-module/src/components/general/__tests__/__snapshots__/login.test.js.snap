// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="app"
  data-testid="login-1"
>
  <header
    className="header"
  >
    <img
      alt="YUMI Logo"
      className="logo"
      src="logo-header.png"
    />
  </header>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      data-testid="loginForm"
    >
      <h3
        className="login-text"
      >
        <strong>
          SIGN IN
        </strong>
      </h3>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="details"
          data-testid="textfields"
        >
          <label
            className="form-label"
          >
            Username
          </label>
          <input
            className="form-control"
            data-testid="emailfield"
            onChange={[Function]}
            placeholder="pizza@gmail.com"
            type="text"
            value=""
          />
        </div>
        <div
          className="details"
        >
          <label
            className="form-label"
          >
            Password
          </label>
          <input
            className="form-control"
            data-testid="passwordfield"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div
          className="details"
          data-testid="signinfield"
        >
          <label
            className="form-label"
          >
            Sign in as:
          </label>
          <select
            aria-label="Default select example"
            className="form-select"
            data-testid="select"
            onChange={[Function]}
            placeholder="User Type"
            type="user type"
            value=""
          >
            <option>
              Choose User Type
            </option>
            <option
              value="Customer"
            >
              Customer
            </option>
            <option
              value="Driver"
            >
              Driver
            </option>
            <option
              value="Restaurant"
            >
              Restaurant
            </option>
          </select>
        </div>
        <button
          className="btn-primary btn btn-primary"
          data-testid="goButton"
          disabled={false}
          type="submit"
        >
           
          Go
           
        </button>
      </form>
    </div>
  </div>
  <div
    className="or"
  >
    <p>
      <strong>
        OR
      </strong>
    </p>
    <p
      className="signupText"
    >
      Sign up as
    </p>
  </div>
  <div
    className="register-buttons"
  >
    <a
      className="btn btn-primary"
      data-testid="customer-signup"
      href="/customer-signup"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Customer
    </a>
     
    <a
      className="btn btn-primary"
      data-testid="driver-signup"
      href="/driver-signup"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Driver
    </a>
     
    <a
      className="btn btn-primary"
      data-testid="restaurant-signup"
      href="/restaurant-regis"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Restaurant
    </a>
  </div>
</div>
`;
